compiler/src/dotty/tools/dotc/reporting/messages.scala

```diff
@@ -3155,3 +3155,42 @@ class VolatileOnVal()(using Context)
 extends SyntaxMsg(VolatileOnValID):
   protected def msg(using Context): String = "values cannot be volatile"
   protected def explain(using Context): String = ""
+
+class UnusedSymbol(pos: SourcePosition,_msg: String, _actions: List[CodeAction])(using Context)
+extends Message(UnusedSymbolID) {
+  def kind = MessageKind.UnusedSymbol
+
+  override def actions(using Context) = _actions
+  override def msg(using Context) = _msg
+  override def explain(using Context) = i""
+}
+
+object UnusedSymbol {
+    private def createRemoveLocalDefAction(tree: tpd.DefTree)(using Context): List[CodeAction] = {
+      import scala.language.unsafeNulls
```

Comment:
This flag disables a way to distinguish typed tree from untyped. We should only use it when we are sure that we won't get them mixed up. I'd rather we use `.nn` where it is necessary.